src/directsketch.rs

```diff
@@ -320,6 +334,33 @@ async fn dl_sketch_assembly_accession(
         file_types = vec![GenBankFileType::Protein];
     }
 
+    let checksums = match download_and_parse_md5(client, &md5sum_url).await {
+        Ok(cs) => cs,
+        Err(err) => {
+            // capture the error message as a string
+            let error_message = err.to_string();
+            // if we can't download/parse the md5sum file, write to checksum failures file to allow manual troubleshooting
+            for file_type in &file_types {
+                // get filename, filetype info to facilitate downstream
+                let url = file_type.url(&base_url, &full_name);
+                let file_name = file_type.filename_to_write(&accession);
+                let failed_checksum_download: FailedChecksum = FailedChecksum {
+                    accession: accession.clone(),
```

Comment:
do these need to be clones?

Reply:
I think you might be able to use [field init shorthand syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#using-the-field-init-shorthand), but I'm not 100% sure how the ownership etc thing works here. So, this is a curious question, rather than requesting :)

Reply:
yes, because I'm using `String`.

If I try to remove the clone:
> move occurs because `accession` has type `std::string::String`, which does not implement the `Copy` trait

I'm open to other solutions, I probably overuse `String`

Reply:
right, I'll think on this and see if I can get anywhere :)

Reply:
if I just use `name` for example instead of `name: name.clone()`, rust-analyzer err:
> consider cloning the value if the performance cost is acceptable: `: name.clone()`

I might be able to avoid the clone by having name be str, not sure

Reply:
I don't think so - I think you need to make it clear that the new struct is taking ownership of the old struct fields. Will have to re-read the code and then see if there's a way to tell Rust to cede ownership!